src/lib.rs

```diff
@@ -134,6 +134,7 @@ pub async fn run_trin(
         portal_jsonrpc_rx,
         state_jsonrpc_tx,
         history_jsonrpc_tx,
```

Comment:
It seems like there's a potential refactor here where we just pass in the header oracle and use the tx channels on that, rather than passing in each channel specifically, but I'll save that for a follow-up pr to avoid cluttering this pr